### PR TITLE
Write librusty_v8.a sha-256 checksum to file.

### PR DESCRIPTION
**Pull request by bnoordhuis**

### CODE REVIEW

**Comment by piscisaureus** [comment]
What's the purpose of this change?
I suspect it has something to do with re-downloading librusty_v8.a after upgrading the rusty_v8 crate.
But there is no way for build.rs to know the "correct" shasum so I don't see how it would work...

**Comment by bartlomieju** [comment]
@piscisaureus https://github.com/denoland/rusty_v8/issues/545

**Comment by piscisaureus** [comment]
@bartlomieju 

There is no way for build.rs to know the "correct" shasum so I don't see how it would work. The only way to find out would be to download the .shasum file on every build, but this will make people that use a mirror unhappy.

My suggestion would be to append the crate version to the filename instead.

**Comment by bartlomieju** [comment]
> The only way to find out would be to download the .shasum file on every build, but this will make people that use a mirror unhappy.

That was the plan to download checksum alongside static lib and verify that checksum of static lib's checksum matches. Probably could skip it for mirrors?

**Comment by piscisaureus** [comment]
@bartlomieju 

Right - so this is not intended to solve the issue that the static library build should be re-downloaded after a rusty_v8 upgrade?

I am aware that some people experience link errors, but that's pretty much always because the download failed entirely and left an empty librusty_v8.a in the build directory. There's no need for shasumming to detect that.

**Comment by bnoordhuis** [comment]
> so this is not intended to solve the issue that the static library build should be re-downloaded after a rusty_v8 upgrade?

No, it's to catch partial or corrupted downloads.

**Comment by bnoordhuis** [comment]
Ping @piscisaureus - your thoughts?

File: Cargo.lock [hunk 1]
```diff
@@ -6,12 +6,52 @@ version = "1.2.1"
...
+
+[[package]]
+name = "generic-array"
+version = "0.14.4"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "501466ecc8a30d1d3b7fc9229b122b2ce8ed6e9d9223f1138d4babb253e51817"
+dependencies = [
+ "typenum",
+ "version_check",
+]
```
Comment (ry) [inline Cargo.lock:53]: It's rather unfortunate that this patch adds a bunch of dependencies just to calculate the checksum. 

https://crates.io/crates/sha1 for example is self-contained. Maybe we could use that - or an already installed command-line utility?

**Comment by ry** [comment]
Closing because stale.

**Pull request closed without merging**